### PR TITLE
Ashima - Solved Issue #7

### LINKED ISSUE #8: Issue 7: Build the Wasm connection (call the wasm version of the compiler instead of the API call to the compiler)
### Is your feature request related to a problem?

This feature will get rid of the requirement of setting up a backend to run the compiler on Compiler-Explorer (CE). Also, it will speed up the compilation process on CE webpage.

### Describe the solution you'd like

1. Compile the compiler to Wasm.
2. Update the API calls so that the front end will call the local wasm compiler code.

### PR DESCRIPTION
**Pull request by ashimapanjwani**
The local wasm version being used in this case was built using the following command
```
wasm-build:
	cargo build --target wasm32-unknown-unknown --features wasm
	wasm-bindgen ./target/wasm32-unknown-unknown/debug/cflat.wasm --out-dir ./cflat/wasm-interface --target web
```

Browser Console Output:

```
compiler-service.ts:218 ============Called submit function==============
compiler-service.ts:232 ============cflat language==============
compiler-service.ts:233 // Type your code here
// By default: load the example named branch.lir

fn g(x:int, y:int, z:int) -> int {
  decl a:int

  entry:
    a = $arith add x y
    a = $arith mul a z
    a = $arith div a 4
    a = $arith add a z
    $ret a
}

hello.js:4 Hello, world!
compiler-service.ts:235 string

**ERROR MESSAGE**

compiler.ts:1398 TypeError: Cannot read properties of undefined (reading '__wbindgen_add_to_stack_pointer')
    at process_string (cflat.js:131:1)
    at CompilerService.<anonymous> (compiler-service.ts:236:39)
    at step (tslib.es6.mjs:147:1)
    at Object.next (tslib.es6.mjs:128:44)
    at tslib.es6.mjs:121:1
    at new Promise (<anonymous>)
    at Module.__awaiter (tslib.es6.mjs:117:1)
    at CompilerService.submit (compiler-service.ts:217:53)
    at Compiler.sendCompile (compiler.ts:1384:14)
    at compiler.ts:1326:22
```

### CODE REVIEW

**Comment by maemre** [comment]
Is this an experimental branch? This doesn't add any functionality besides adding any utility to CE.

Also, use more descriptive names than `process_string` for the entry point of the compiler.

**Comment by ashimapanjwani** [comment]
@maemre I was able to get the assembly code to show up in the UI output window after passing the assembly code to the `processAsm` function. 
The `processAsm` function is more or less copied from `lib/parsers/asm-parser.ts`.
This is a sample result object that I'm returning as part of the response
```json
{
    "source": "LIR source code",
    "asm": [
        {
            "text": "_f:",
            "source": null,
            "labels": []
        }
    ],
    "labelDefinitions": {
        "_f": 1,
        "f_entry": 7,
        "f_f": 15,
        "f_t": 18,
        "f_epilogue": 22
    },
    "filteredCount": 18
}
```

Just wanted your feedback.

File: static/process-asm.ts [hunk 1]
```diff
@@ -0,0 +1,548 @@
+import {ParseFiltersAndOutputOptions} from '../types/features/filters.interfaces.js';
```
Comment (maemre) [inline static/process-asm.ts:5]: please add a comment to indicate where this file comes from. Is it copied from somewhere in the backend?

**Review by maemre — CHANGES_REQUESTED** [review]

**Comment by maemre** [comment]
That looks reasonable to me (it does process the x86 assembly labels correctly). I just asked for adding the provenance of the `process-asm.ts` file as a comment in the beginning of the file. Or, if you can import where that file originally is, that's even better.

File: static/process-asm.ts (hunk 1 above)
Comment (ashimapanjwani) [reply static/process-asm.ts:5]: Addressed

**Comment by ashimapanjwani** [comment]
> That looks reasonable to me (it does process the x86 assembly labels correctly). I just asked for adding the provenance of the `process-asm.ts` file as a comment in the beginning of the file. Or, if you can import where that file originally is, that's even better.

@maemre I added a comment at the beginning of `process-asm.ts` to explain where the code is originally taken from. 

Initially, I had tried importing and trying to reuse the code from the original file `compiler-explorer/lib/parsers/asm-parser.ts` but some of the node libraries were throwing an error as they did not work on the client side. The fix involved making changes to the `tsconfig` and `webpack.config` files but it soon got very complicated and despite making changes, I was still getting different errors. So copying the backend code to the front end with a few minor modifications to get it to work on the client side seemed to be the more elegant solution in this case (despite the code duplication).

**Review by maemre — APPROVED** [review]

**Pull request merged**